cpuid/src/lib.rs

```diff
@@ -61,11 +62,18 @@ pub fn filter_cpuid(
     ht_enabled: bool,
     kvm_cpuid: &mut CpuId,
 ) -> Result<(), Error> {
+    let vendor_id = get_vendor_id().map_err(Error::InternalError)?;
     let vm_spec = VmSpec::new(cpu_id, cpu_count, ht_enabled);
 
-    let transform_entry_fn: EntryTransformerFn = intel::transform_entry;
+    let cpuid_transformer: &dyn CpuidTransformer = match &vendor_id {
+        INTEL => &intel::IntelCpuidTransformer {},
+        AMD => &amd::AmdCpuidTransformer {},
+        _ => &other::OtherCpuidTransformer {},
```

Comment:
What is the purpose of having `OtherCpuidTransformer`? When we add a new platform shouldn't we just add the corresponding transformer and not have a generic one here?

Reply:
I removed it